src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs

```diff
@@ -329,8 +327,33 @@ public void Dispose()
 
         private void DisposeCache()
         {
-            _cached?.Dispose();
-            _cached = null;
+            //If _cached is not null - it means that the client approached with this multitask request to node and the request failed.
+            //and now client tries to send it to another node.
+            if (_cached != null)
+            {
+                var cachedValues = _cached.Values;
+                if (cachedValues != null)
+                {
+                    foreach (var cached in cachedValues)
+                    {
+                        cached.Release.NotModified();
```

Comment:
Wait, in here, didn't have to take into account that we got an error from the server?
In that case, we don't want this, do we?

Reply:
Right, I reverted this part because it was fixed in RavenDB-16358 issue.